ghostbuster/src/main/scala/not/ogame/bots/ghostbuster/GBot.scala

```diff
@@ -0,0 +1,33 @@
+package not.ogame.bots.ghostbuster
+
+import java.time.{Clock, Instant, LocalDateTime}
+
+import com.softwaremill.quicklens._
+import not.ogame.bots.facts.SuppliesBuildingCosts
+
+class GBot(jitterProvider: RandomTimeJitter)(implicit clock: Clock) {
+  def nextStep(state: State): State = {
+    state.wishList match {
+      case (wish: Wish.Build) :: tail if state.scheduledWishes.isEmpty =>
+        val requiredResources = SuppliesBuildingCosts.buildingCost(wish.suppliesBuilding, wish.level)
+        if (state.suppliesPage.currentResources.gtEqTo(requiredResources)) {
+          scheduleWish(state, wish, tail, clock.instant())
+        } else {
+          val difference = state.suppliesPage.currentResources.difference(requiredResources)
+          val hoursToWait = difference.div(state.suppliesPage.currentProduction).max
```

Comment:
Difference is using abs which can cause miscalculation here. If you have significantly more metal then required but you lack a little crystal you will have wrongly high hoursToWait.

Reply:
Good catch, fixed.